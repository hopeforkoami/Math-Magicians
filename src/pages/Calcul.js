import React from 'react';
import Calculator from '../components/Calculator';

function Calcul() {
  return (
    <div className="calcul_page">
      <h1 className="calcul_page">
        Let&apos;s do some math
      </h1>
      <Calculator />
    </div>
  );
}

export default Calcul;
