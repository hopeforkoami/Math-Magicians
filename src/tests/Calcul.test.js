import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Calcul from '../pages/Calcul';

test('render Calcul component', () => {
  const { asFragment } = render(<Router><Calcul /></Router>);
  expect(asFragment()).toMatchSnapshot();
});
