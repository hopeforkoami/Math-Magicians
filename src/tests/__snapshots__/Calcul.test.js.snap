// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Calcul component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header>
      <div
        class="project_title"
      >
        Math Magicians
      </div>
      <nav>
        <ul
          class="menu"
        >
          <a
            href="/"
          >
            <li
              class="menu_element start_menu_element selected_menu_element"
            >
              Home
            </li>
          </a>
          <a
            href="/calcul"
          >
            <li
              class="menu_element"
            >
              Calculator
            </li>
          </a>
          <a
            href="/quote"
          >
            <li
              class="menu_element"
            >
              Quote
            </li>
          </a>
        </ul>
      </nav>
    </header>
    <section
      class="section calculator_section"
    >
      <h1>
        Let's do some math!
      </h1>
      <div
        class="calculator section_content"
      >
        <div>
          <div
            class="calc-container"
          >
            <table
              class="calculator"
            >
              <tbody>
                <tr>
                  <td
                    class="result"
                    colspan="4"
                  >
                    0
                  </td>
                </tr>
                <tr>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="AC"
                    >
                      AC
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="+/-"
                    >
                      +/-
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="%"
                    >
                      %
                    </button>
                  </td>
                  <td
                    class="action"
                  >
                    <button
                      type="button"
                      value="÷"
                    >
                      ÷
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="7"
                    >
                      7
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="8"
                    >
                      8
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="9"
                    >
                      9
                    </button>
                  </td>
                  <td
                    class="action"
                  >
                    <button
                      type="button"
                      value="x"
                    >
                      x
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="4"
                    >
                      4
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="5"
                    >
                      5
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="6"
                    >
                      6
                    </button>
                  </td>
                  <td
                    class="action"
                  >
                    <button
                      type="button"
                      value="-"
                    >
                      -
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="1"
                    >
                      1
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="2"
                    >
                      2
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="3"
                    >
                      3
                    </button>
                  </td>
                  <td
                    class="action"
                  >
                    <button
                      type="button"
                      value="+"
                    >
                      +
                    </button>
                  </td>
                </tr>
                <tr>
                  <td
                    class="num"
                    colspan="2"
                  >
                    <button
                      type="button"
                      value="0"
                    >
                      0
                    </button>
                  </td>
                  <td
                    class="num"
                  >
                    <button
                      type="button"
                      value="."
                    >
                      .
                    </button>
                  </td>
                  <td
                    class="action"
                  >
                    <button
                      type="button"
                      value="="
                    >
                      =
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`render Quote component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header>
      <div
        class="project_title"
      >
        Math Magicians
      </div>
      <nav>
        <ul
          class="menu"
        >
          <a
            href="/"
          >
            <li
              class="menu_element start_menu_element selected_menu_element"
            >
              Home
            </li>
          </a>
          <a
            href="/calcul"
          >
            <li
              class="menu_element"
            >
              Calculator
            </li>
          </a>
          <a
            href="/quote"
          >
            <li
              class="menu_element"
            >
              Quote
            </li>
          </a>
        </ul>
      </nav>
    </header>
    <section
      class="section quote_section"
    >
      <div
        class="section_content"
      >
        <p
          class="quote_citation"
        >
          Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston.
        </p>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
